docs/source/format/Columnar.rst

```diff
@@ -487,6 +499,103 @@ will be represented as follows: ::
           |-------------------------------|-----------------------|
           | 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 | unspecified (padding) |
 
+ListView Layout
+~~~~~~~~~~~~~~~
+
+The ListView layout is defined by three buffers: a validity bitmap, an offsets
+buffer, and an additional sizes buffer. Sizes and offsets have the identical bit
+width and both 32-bit and 64-bit signed integer options are supported.
+
+As in the List layout, the offsets encode the start position of each slot in the
+child array. In contrast to the List layout, list lengths are stored explicitly
+in the sizes buffer instead of inferred. This allows offsets to be out of order.
+Elements of the child array do not have to be stored in the same order they
+logically appear in the list elements of the parent array.
+
+Every list-view value, including null values, has to guarantee the following
+invariants: ::
+
+    0 <= offsets[i] <= length of the child array
+    0 <= offsets[i] + size[i] <= length of the child array
+
+A list-view type is specified like ``ListView<T>``, where ``T`` is any type
+(primitive or nested). In these examples we use 32-bit offsets and sizes where
+the 64-bit version would be denoted by ``LargeListView<T>``.
+
+**Example Layout: ``ListView<Int8>`` Array**
+
+We illustrate an example of ``ListView<Int8>`` with length 4 having values::
+
+    [[12, -7, 25], null, [0, -127, 127, 50], []]
```

Comment:
It would be nice to have an example that shows what happens with duplicate lists and duplicate values:

```
[[12, -7, 25], null, [0, -127, 127, 12], [], [12, -7, 25]]
```

Reply:
@adriangb anything can happen: they can be duplicated in the data or entries can point to the same data.

Compact representation:

```
buffers:
  offsets: [0, _, 3, _, 0]
  sizes:   [3, _, 4, 0, 3]

children:
  values: [12, -7, 25, 0, -127, 127, 12]
```

Common representation:

```
buffers:
  offsets: [0, _, 3, _, 7]
  sizes:   [3, _, 4, 0, 3]

children:
  values: [12, -7, 25, 0, -127, 127, 12, 12, -7, 25]
```

Reply:
Makes total sense, thanks!